front/ticketcontract.form.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * ---------------------------------------------------------------------
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2015-2021 Teclib' and contributors.
+ *
+ * http://glpi-project.org
+ *
+ * based on GLPI - Gestionnaire Libre de Parc Informatique
+ * Copyright (C) 2003-2014 by the INDEPNET Development Team.
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * GLPI is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * GLPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with GLPI. If not, see <http://www.gnu.org/licenses/>.
+ * ---------------------------------------------------------------------
+ */
+
+include ('../inc/includes.php');
+
+Session ::checkLoginUser();
+
+$item = new TicketContract();
+
+if (isset($_POST["add"])) {
+   if (!empty($_POST['tickets_id']) && empty($_POST['contracts_id'])) {
+      $message = sprintf(
+         __('Mandatory fields are not filled. Please correct: %s'),
+         Contract::getTypeName(1)
+      );
+      Session::addMessageAfterRedirect($message, false, ERROR);
+      Html::back();
+   }
+   if (empty($_POST['tickets_id']) && !empty($_POST['contracts_id'])) {
+      $message = sprintf(
+         __('Mandatory fields are not filled. Please correct: %s'),
+         Ticket::getTypeName(1)
+      );
+      Session::addMessageAfterRedirect($message, false, ERROR);
+      Html::back();
+   }
```

Comment:
This is pasted code from similar links, you can find it in:
* `change_ticket.form.php`
* `contract_item.form.php`
* `itm_softwarelicense.form.php`
* `item_ticket.form.php`
* `problem_ticket.form.php`
* `ticketcontract.form.php`

If we want to do things differently it should be updated everywhere in another PR (also merging all the code in one place to avoid pasting the same thing over and over would be nice...)